### ISSUE TITLE
Target version 1.0.3 of the shared framework by default

### ISSUE THREAD

**Issue by dsplaisted** [dotnet/sdk#572]
Projects should by default target the latest patch version of the shared framework for the version of .NET Core they are targeting.  So (currently) an app targeting netcoreapp1.0 should target version 1.0.3 of the shared framework.

In order to do this, we need to make sure that the tooling (Visual Studio, the .NET CLI) installs this runtime.  We'll also need to update our CI servers to include it.

After #450 is merged, changing this would involve updating `Microsot.NET.Sdk.BeforeCommon.targets`.

**Comment by srivatsn** [comment]
@mlorbetske mentioned that they've already done this with the asp.net template. I assume that means that VS and CLI both already have the 1.0.3 shared framework? For the CI servers, we just need to move the latest CLI then if the CLI already has 1.0.3. @piotrpMSFT @livarcocc
